### PR TITLE
[full-ci] Redesign FilesTable

### LINKED ISSUE #6207: Redesign "Files Table"
# Status

# Description

## User Stories

* > As a user I want the files table to be presented in an appealing and well structured way so that my cognitive costs are  low. 

## Value

- enhance filebrowsing experience

## Acceptance Criteria (Desktop)
- Header: sortable, arrow points up/down if the column is the current "sorted-by" column; on hover (only for sortable columns): arrow (fill type: line) shows up to indicate, that the column is sortable. see attached gifs. 
- Quickactions: 3-dots always visible
- Quickactions: other quickactions appear on hover 
- file icon is clickable (not only the filename)
- (nice to have 🍰) quickactions: copy link changes icon for short time, if it got clicked (cf. right sidebar -> copy public link) see attached gifs.

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] codre review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

image

image

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See https://github.com/owncloud/web/issues/6207

## Related Issue

- Fixes #6207 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests updated
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-runtime/themes/owncloud/theme.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
         "background-default": "#ffffff",
         "background-highlight": "#edf3fa",
         "background-muted": "#f8f8f8",
+        "background-secondary": "#ECECEC",
```
Comment (kulmann) [inline packages/web-runtime/themes/owncloud/theme.json:23]: Please keep the hex codes all lowercase for consistency. Also need to update them in the docs.

Comment (kulmann) [reply packages/web-runtime/themes/owncloud/theme.json:23]: Also, please make sure that that you make similar adjustments to the dark theme color palette.

File: packages/web-app-files/src/components/FilesList/QuickActions.vue [hunk 2]
```diff
@@ -39,3 +39,12 @@ export default {
   }
 }
 </script>
+
+<style lang="scss">
+.quick-action-button {
+  padding: 6px !important;
```
Comment (kulmann) [inline packages/web-app-files/src/components/FilesList/QuickActions.vue:45]: Why the hardcoded pixel value for the padding? You can do something like `calc(1.5 * var(--oc-space-xsmall))`.

**Comment by kulmann** [comment]
Could you please rebase this PR at some point? It's based on a pretty old commit. 😉

File: packages/web-app-files/src/components/FilesList/QuickActions.vue (hunk 2 above)
Comment (lookacat) [reply packages/web-app-files/src/components/FilesList/QuickActions.vue:45]: yes that better 👍🏼  couldnt think of this solution

File: packages/web-runtime/themes/owncloud/theme.json (hunk 1 above)
Comment (lookacat) [reply packages/web-runtime/themes/owncloud/theme.json:23]: alright

**Comment by lookacat** [comment]
@kulmann yes will do a rebase before using the alpha ODS RC

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/22929/66/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:94

File: changelog/unreleased/enhancement-redesign-filestable [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Redesign FilesTable
+
+We've adjusted the QuickActions and updated theming
```
Comment (fschade) [inline changelog/unreleased/enhancement-redesign-filestable:3]: i would expect a bit more detail on quickAction changes.

**Review by fschade — APPROVED** [review]
just one small thing in the changelog, expect that LGTM 🚀

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/22936/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesList-sort_feature-L51.png
  

### webUIFilesList-sort_feature-L72.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/22936/56/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesList-sort_feature-L51.png
  

### webUIFilesList-sort_feature-L72.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/22939/55/1
The following scenarios passed on retry:
- webUIMarkdownEditor/markdownFile.feature:21

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/22939/69/1
The following scenarios passed on retry:
- webUIUserJourney/journey1.feature:11

**Pull request merged**